### PR TITLE
feat: extends roles for the controllers

### PR DESCRIPTION
**Pull request by bartoszmajsak**
This is needed to make our enhancements based on resource annotations to work.

Marking it as a draft - this still consists of PR-specific images of both ODH Dashboard and Controller

### CODE REVIEW

File: enabling-ossm.md [hunk 1]
```diff
@@ -75,7 +75,7 @@ kubectl wait --for condition=Ready smcp/basic --timeout 300s -n istio-system
 ### Create Kubeflow Definition
 
 ```sh
-cat <<'EOF' > odh-minimal.yaml
+cat <<'EOF' > odh-minimal.ign.yaml
```
Comment (bartoszmajsak) [inline enabling-ossm.md:78]: This way you can have global `.gitignore` config to ignore files matching e.g. pattern `*.ign.*`

File: odh-notebook-controller/odh-notebook-controller/base/kustomization.yaml [hunk 2]
```diff
@@ -3,6 +3,6 @@ kind: Kustomization
 resources:
 - ../default
 images:
-- name: quay.io/opendatahub/odh-notebook-controller
-  newName: quay.io/maistra-dev/odh-notebook-controller
-  newTag: enable_ossm
+  - name: quay.io/opendatahub/odh-notebook-controller
+    newName: quay.io/bmajsak/odh-notebook-controller
+    newTag: ossm_annotations
```
Comment (bartoszmajsak) [inline odh-notebook-controller/odh-notebook-controller/base/kustomization.yaml:8]:
```suggestion
- name: quay.io/opendatahub/odh-notebook-controller
  newName: quay.io/bmajsak/odh-notebook-controller
  newTag: ossm_annotations
```

File: odh-notebook-controller/odh-notebook-controller/base/kustomization.yaml [hunk 3]
```diff
@@ -4,5 +4,5 @@ resources:
 - ../default
 images:
 - name: quay.io/opendatahub/odh-notebook-controller
-  newName: quay.io/maistra-dev/odh-notebook-controller
-  newTag: enable_ossm
+  newName: quay.io/bmajsak/odh-notebook-controller
+  newTag: ossm_annotations
```
Comment (cam-garrison) [inline odh-notebook-controller/odh-notebook-controller/base/kustomization.yaml:8]:
```suggestion
  newName: quay.io/maistra-dev/odh-notebook-controller
  newTag: enable_ossm
```

**Review by cam-garrison — APPROVED** [review]
Bar my suggestion, LGTM! We will sort out which `odh-dashboard` image to use in a future PR.

**Pull request merged**